index.js

```diff
@@ -8,7 +8,10 @@ var CacheStore = require('./cache_store');
 var DEFAULT_TOTAL_VNODES = 1024;
 var DEFAULT_CALM_THRESHOLD = 500;
 var DEFAULT_RETRY_INTERVAL_MS = 5000;
-var MAX_PARALLEL_TASKS = 10;
+
+// We're setting a modest limit to make sure we're not starving the event-loop of CPU
+// with overly aggressive setting
+var MAX_PARALLEL_TASKS = 2;
```

Comment:
This var is used by other code path as well, is that we want as well?

Reply:
Yes, limiting the concurrency across the board